src/integrations/pathManager.ts

```diff
@@ -0,0 +1,108 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import paths from '@/utils/paths';
+import { manageLinesInFile } from '@/integrations/manageLinesInFile';
+
+let pathManager: ManualPathManager;
+
+// PathManager is the interface that anything that manages the
+// PATH variable must implement.
+interface PathManager {
+  // Makes real any changes to the system. Should be idempotent.
+  enforce(): Promise<void>
+  // Removes any changes that the PathManager may have made.
+  // Should be idempotent.
+  remove(): Promise<void>
+}
+
+// ManualPathManager is for when the user has chosen to manage
+// their PATH themselves. It does nothing.
+class ManualPathManager implements PathManager {
+  async enforce(): Promise<void> {}
+  async remove(): Promise<void> {}
+}
+
+// RcFilePathManager is for when the user wants Rancher Desktop to
+// make changes to their PATH by putting lines that change it in their
+// shell .rc files.
+export class RcFilePathManager implements PathManager {
+  constructor() {
+    const platform = os.platform();
+
+    if (platform !== 'linux' && platform !== 'darwin') {
+      throw new Error(`Platform "${ platform }" is not supported by RcFilePathManager`);
+    }
+  }
+
+  async enforce(): Promise<void> {
+    await this.managePosix(true);
+    await this.manageCsh(true);
+    await this.manageFish(true);
+  }
+
+  async remove(): Promise<void> {
+    await this.managePosix(false);
+    await this.manageCsh(false);
+    await this.manageFish(false);
+  }
+
+  protected async managePosix(desiredPresent: boolean): Promise<void> {
+    const pathLine = `PATH=\${PATH}:${ paths.integration }`;
+
+    await Promise.all(['.bashrc', '.zshrc'].map((rcName) => {
+      const rcPath = path.join(os.homedir(), rcName);
+
+      return manageLinesInFile(rcPath, [pathLine], desiredPresent);
+    }));
+  }
+
+  protected async manageCsh(desiredPresent: boolean): Promise<void> {
+    const pathLine = `set path=($path ${ paths.integration })`;
```

Comment:
I haven't used `csh` in decades; how do they deal with spaces in paths?

Reply:
It seems that setting `path` is fundamentally flawed with respect to spaces:

```
% echo $path
/sbin /usr/sbin /bin /usr/bin /usr/local/bin /usr/local/sbin /opt /home/adam/bin /home/adam/.local/bin /usr/local/go/bin .
% set path=("$path" 'with spaces')
% echo $path
/sbin /usr/sbin /bin /usr/bin /usr/local/bin /usr/local/sbin /opt /home/adam/bin /home/adam/.local/bin /usr/local/go/bin . with spaces
% echo $PATH
/sbin /usr/sbin /bin /usr/bin /usr/local/bin /usr/local/sbin /opt /home/adam/bin /home/adam/.local/bin /usr/local/go/bin .:with spaces
% set path=("$path" 'another one')
% echo $path
/sbin /usr/sbin /bin /usr/bin /usr/local/bin /usr/local/sbin /opt /home/adam/bin /home/adam/.local/bin /usr/local/go/bin . with spaces another one
% echo $PATH
/sbin /usr/sbin /bin /usr/bin /usr/local/bin /usr/local/sbin /opt /home/adam/bin /home/adam/.local/bin /usr/local/go/bin . with spaces:another one
```

With this, each time `path` is set the previous `path` elements get turned into one big element of `path`. Not good. The following was enlightening:

```
% echo $path
/sbin /usr/sbin /bin /usr/bin /usr/local/bin /usr/local/sbin /opt /home/adam/bin /home/adam/.local/bin /usr/local/go/bin .
% echo $PATH
/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/bin:/usr/local/sbin:/opt:/home/adam/bin:/home/adam/.local/bin:/usr/local/go/bin:.
% setenv PATH $PATH\:"with spaces"
% echo $path
/sbin /usr/sbin /bin /usr/bin /usr/local/bin /usr/local/sbin /opt /home/adam/bin /home/adam/.local/bin /usr/local/go/bin . with spaces
% echo $PATH
/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/bin:/usr/local/sbin:/opt:/home/adam/bin:/home/adam/.local/bin:/usr/local/go/bin:.:with spaces
% setenv PATH $PATH\:"more spaces"
setenv: Too many arguments.
% setenv PATH "$PATH"\:"more spaces"
% echo $PATH
/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/bin:/usr/local/sbin:/opt:/home/adam/bin:/home/adam/.local/bin:/usr/local/go/bin:.:with spaces:more spaces
% echo $path
/sbin /usr/sbin /bin /usr/bin /usr/local/bin /usr/local/sbin /opt /home/adam/bin /home/adam/.local/bin /usr/local/go/bin . with spaces more spaces
```

So, `setenv PATH "$PATH"\:"ourPath"` should work.